Kernel/Types/analyze.cpp

```diff
@@ -514,20 +514,23 @@ static string hundreds[10]= {"",  "c",  "cc",  "ccc",  "cd",
                              "d", "dc", "dcc", "dccc", "cm"};
 
 string
-roman_nr (int nr) {
-  if (nr < 0) return "-" * roman_nr (-nr);
+roman_nr (int32_t nr) {
   if (nr == 0) return "o";
-  if (nr > 1000) return "m" * roman_nr (nr - 1000);
   if (nr == 1000) return "m";
   if (nr == 999) return "im";
   if (nr == 499) return "id";
+  if (nr >= 3999) return "<unspecified>";
+  if (nr <= -3999) return "<unspecified>";
```

Comment:
```suggestion
  if (nr > 3999) return "<unspecified>";
  if (nr < -3999) return "<unspecified>";
```
result of 3999 is correct

Reply:
is `mmmim` correct?

Reply:
via googling `3999 in roman numerals`, this is the answer MMMCMXCIX

Reply:
well 999 in the current impl is not correct

Reply:
3999 is the maximum number, that's the truth via google.